### PR TITLE
feat(Datagrid): call onClearFilters when clearing filters from tag summary

### PR DESCRIPTION
**Pull request by xmannyxfreshx1**
Closes https://github.com/carbon-design-system/ibm-products/issues/5879

support passing a callback function to the filter prop `onClearFilters`. Then if it's not undefined and a function, call it when clicking the 'Clear filters' button in the filter summary component.

#### What did you change?
```
packages/ibm-products/src/components/Datagrid/Datagrid/DatagridContent.tsx
packages/ibm-products/src/components/Datagrid/types/index.ts
```

#### How did you test and verify your work?
Storybook

### CODE REVIEW

**Review by devadula-nandan — CHANGES_REQUESTED** [review]
Thanks for the contribution @xmannyxfreshx1,
could you update the story docs, by mentioning this callback usage?
The code looks good to me.

File: packages/ibm-products/src/components/Datagrid/Datagrid/DatagridContent.tsx [hunk 1]
```diff
@@ -217,7 +217,15 @@ export const DatagridContent = ({
       <FilterSummary
         className={`${blockClass}__filter-summary`}
         filters={filterTags}
-        clearFilters={() => EventEmitter.dispatch(CLEAR_FILTERS, tableId)}
+        clearFilters={() => {
+          EventEmitter.dispatch(CLEAR_FILTERS, tableId);
+          if (
+            typeof filterProps?.onClearFilters !== 'undefined' &&
```
Comment (devadula-nandan) [inline packages/ibm-products/src/components/Datagrid/Datagrid/DatagridContent.tsx:223]: I think this is not necessary.
``` suggestion
```

**Review by devadula-nandan — CHANGES_REQUESTED** [review]

**Comment by xmannyxfreshx1** [comment]
@devadula-nandan Thanks for the feedback, I've made your suggested changes.

**Review by devadula-nandan — APPROVED** [review]
LGTM

**Review by anamikaanu96 — APPROVED** [review]
Hey @xmannyxfreshx1 Thank you for your contribution! 
The code looks fine. Approved. 🎉

**Pull request merged**